### PR TITLE
First commit: nicely styled, WIP

### PR DESCRIPTION
**Pull request by callumflack**
1st draft, incomplete UI design. To note:

1. The "save" instructions is a placeholder div for now. See it commented out at App/App.js:69

2. The notebook filters list is also commented out for now. See Notes/Notebooks.js:91

3. I use Prettier in VSCode. I frogot to turn it off before working, so it's done some reformatting… Should we decide on via a .prettierrc or should I turn it off?

4. As discussed with main menu toggling filters / instructions, could you also please add an active class for the active button? 

5. This is my first go at OSS! I've prob not got this PR quite right. I'm all ears as we go thru this.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,15 +31,16 @@
...
+    "dev": "npm run start"
   },
   "devDependencies": {
     "path": "^0.12.7",
     "react-dev-utils": "^3.1.1",
     "react-scripts": "0.9.5"
   },
   "engines": {
-    "node": "8.x"
+    "node": "^10.3.0"
```
Comment (galtenberg) [inline package.json]: Important to do in this particular PR? If so, I'll probably go through a full upgrade _before_ merging this, just to get everything up to date.

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,48 +1,89 @@
-import React, { Component } from 'react'
-import { Link } from 'react-router-dom'
-import classnames from 'classnames'
-import './style.css'
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import classnames from "classnames";
+import "./style.css";
```
Comment (galtenberg) [inline src/components/App/App.js]: Are the double quote changes important?

File: src/components/App/App.js [hunk 3]
```diff
@@ -1,48 +1,89 @@
...
 
-import Notebooks from '../Notes/Notebooks'
+import Notebooks from "../Notes/Notebooks";
```
Comment (galtenberg) [inline src/components/App/App.js]: I don't use semicolons in js, so can this prettyfier rule be removed and undone?

**Comment by galtenberg** [comment]
Great contribution, man - seriously - 

I made a few specific prettier comments - it is always easier if things like that are done in separate PRs, so that we can easily see material differences. If those can be all the way backed out, we can chat this out later. See how much time it'd take either way (backing those changes out, vs. discussing specific points).

I'll have to look into how glitch plays with yarn files.

File: package.json (hunk 1 above)
Comment (callumflack) [reply package.json]: I couldn't get it to run locally w/o upgrading it.

Saves me having to remember adding the line if it's not, dare I say, a big thing to do a full upgrade… :) but if it is, I'll just not commit it.

File: src/components/App/App.js (hunk 2 above)
Comment (callumflack) [reply src/components/App/App.js]: No, just my Prettier defaults (apologies, it also reformatted a fair bit). I think we can add a .prettierrc that handles formatting per project. Will add.

File: src/components/App/App.js (hunk 3 above)
Comment (callumflack) [reply src/components/App/App.js]: Yes, certainly. Will do as part of prev comment.

**Comment by callumflack** [comment]
Cheers Chris! 

If yarn doesn't play with Glitch, can use npm, no fuss.

**Comment by galtenberg** [comment]
Cool man - I'm actually testing package upgrades based on `10.x`, I'll let you know how it goes

File: .gitignore [hunk 4]
```diff
@@ -11,6 +11,7 @@
 
 # misc
 .DS_Store
+.env
```
Comment (galtenberg) [inline .gitignore]: Also add `yarn.lock` to this file?

**Comment by galtenberg** [comment]
Do add the prettier config, and I'll see if I can get my vim to auto apply it.

And/or maybe also add a linter that agrees with the prettier config?

File: .gitignore (hunk 4 above)
Comment (callumflack) [reply .gitignore]: D'oh, added

**Comment by callumflack** [comment]
Re: eslint + prettier, here's a concise rundown of how CRA do it: https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#formatting-code-automatically

I can then simply add our rules in the `lint-staged` object (single quotes / no semis).

Good for you?

I took the time to dig a little because eslint + prettier can get it messy pretty quickly. Turns out eslint is already w/i CRA, so I'm glad I checked. CRA handles it nicely.

**Comment by galtenberg** [comment]
@callumflack This sounds just right to me 👍 :beer:

File: .gitignore [hunk 5]
```diff
@@ -11,12 +11,14 @@
...
 .env.local
 .env.development.local
 .env.test.local
 .env.production.local
 
 npm-debug.log*
+yarn.lock
```
Comment (galtenberg) [inline .gitignore]: @callumflack So I think the only thing remaining then is to remove the actual yarn.lock from the PR, probably via `git rm yarn.lock`, commit/push, then put it right back locally and git should ignore it from here on.

Comment (callumflack) [reply .gitignore]: Again, d'oh. Pushed ;)

**Comment by callumflack** [comment]
FYI: the node engines problem is easily countered by adding the `--ignore-engines` flag during installs

**Comment by callumflack** [comment]
@galtenberg OK, added that nice little lint-staged script. That's a much better way of handling formatting than managing 2 x config files for eslint & prettier. I put off figuring out the perfect solution for that for months.

Can you just check the globbing for me?

File: .vscode/settings.json [hunk 6]
```diff
@@ -0,0 +1,4 @@
+{
+  "prettier.semi": false,
+  "prettier.singleQuote": true
+}
```
[the remark this answers is not in the document]
Comment (callumflack) [reply .vscode/settings.json]: Yes, certainly. 

I have a limited git vocab. I usually don't work w/ anyone but myself on code… These little pointers help, ta.

[the remark this answers is not in the document]
Comment (callumflack) [reply .vscode/settings.json]: For some reason running `git rm .vscode/settings.json` does this for me?

Total basic stuff here. Sorry.

image

[the remark this answers is not in the document]
Comment (callumflack) [reply .vscode/settings.json]: Hmmm, wrapping the path in quotes worked. But it deletes it not just from git, but entirely. Would you know why? 

I'll leave it out. The `lint-staged` on commit should suffice.

[the remark this answers is not in the document]
Comment (galtenberg) [reply .vscode/settings.json]: Yeah, sorry that's normal git behavior, I should have suggested `git rm --cached [filename]`.

[the remark this answers is not in the document]
Comment (callumflack) [reply .vscode/settings.json]: Thanks!

File: src/components/App/Header.js [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+              Randomnote
+              <IconRandom />
+            </a>
+          </div>
+          <div className="f Header-buttons">
+            <div>
+              <a onClick={this.props.notebooksToggle}>
+                <button>
+                  <IconFilter />Filter
+                </button>
```
Comment (galtenberg) [inline src/components/App/Header.js]: Wondering if we could have a single `<SvgIconButton>` component, that accepts props of `width`, `height`, `title`, `iconPath`, `fillPath`.

**Pull request merged**